### ISSUE TITLE
Searching using coordinates is useless as markers will be put with a large error margin

### ISSUE THREAD

**Issue by piiskop** [nextcloud/maps#419]
Tonight, I participated on an adventure game. We knew ahead the coordinates of six control points. So I wanted to use my new toy _Nextcloud Maps_ for that. As I was running already, it turned out that almost all of these six pair of coordinates showed up incorrect results. For instance: 59.39736,24.29106. The marker will be set to Merenuka in the forest near a road. Actually, that point is like half a kilometer South. I fulfilled with that information my story on [setting up FOSS-directions](https://medium.com/arvutiteadus/foss-driving-directions-in-nextcloud-5ff23a87f83c?postPublishedType=repub) where you can see screenshots, too.

**Comment by tacruc** [comment]
I guess the curtial infromation is that a search for `59.39736,24.29106` leads to marker at `"lat":"59.4012388","lon":"24.2897721"`

**Comment by piiskop** [comment]
Are you a developer or just somebody that guesses? Developers should 
know exactly what they do. I see that you've contributed something to 
this project, so you should know, not guess. What is curtial information 
anyway? /Maps/ should result correct locations not some /curtial stuff/ 
to be useful.

*peacecop kalmer:*

*Instituudi tee 3-27 *

*76902 * *Harku *

*Harku*

*Harjumaa*
Estonia

*+372 652 4228 *

*+372  5620 4556 *

*__________________________________________________________________ *

**Comment by tacruc** [comment]
> Are you a developer or just somebody that guesses? 

Right now I'm just somebody that guesses. If this turns out to be a nice conversation, and it turns out that I understand whats your problem, It might be, that I'm the guy who uses his free time to fix you problem. 

So right now I'm trying to understand what you want, without reading 4 pages about how to install stuff.

So in short you search for: `59.39736,24.29106`
and the marker result is not as expected at `59.39736,24.29106`, but `"lat":"59.4012388","lon":"24.2897721"`

Or did I miss anything?

**Comment by piiskop** [comment]
You're right - the marker's location is totally incorrect. Somebody else 
already explained that /Nextcloud Maps/ is looking for the closest named 
location. Why does one need a closest named location if that one knows 
the coordinates? If coordinates are given why not to show the exact 
location and if an address or a name is given then show the location for 
that name or if ambiguous, display a list with possible locations! I 
programmed a similar thing more than a decade ago for a weather 
application. One could just point a location on the map and the list 
with nearest locations would be displayed. Those locations were 
necessary for getting the weather data as that was connected to those 
named locations, not exact coordinates. /Nextcloud Maps/ doesn't display 
any information that's connected to named locations, so why doesn't a 
search with coordinates result with a marker on the exact location? If I 
put a marker using my pointing device then the marker will be set in the 
exact location, not a nearest named location. Isn't it a contradiction? 
If somebody gives you coordinates for a treasure then you want to go 
that exact location, right? Not to a location that's like a kilometer away.

*peacecop kalmer:*

*Instituudi tee 3-27 *

*76902 * *Harku *

*Harku*

*Harjumaa*
Estonia

*+372 652 4228 *

*+372  5620 4556 *

*__________________________________________________________________ *

**Comment by piiskop** [comment]
As I'm kind of calmed down meanwhile, I gave myself a try to reread your response. You made a grammatical mistake with the word _curtial_ that doesn't exist. So I understood that you meant _curtail_ and that your guess was on the software. However, now I understand that you meant _crutial_ and therefore your guess was on my report not the software.

Meanwhile a Nominatim specialist mtmail explained what Nominatim is for - it's for getting names as the name of the software is. So this issue is directly related to _Nextcloud Maps_.

**Comment by Spartachetto** [comment]
@tacruc given the specific behaviour of Nominatim, there are three possibilities, I think. The first one is to maintain the current behaviour; the second one is to select the exact coordinates if the user inputs them; the third one is to provide the user an option (something like "do you want to select the exact coordinates or the nearest named place?"). It is clear that you could choose to change the selected behaviour according to the input (something like "if there are enough decimal digits pick the exact location, otherwise ask"). This is something related to the preferred UI, as I know that in general Nextcloud prefers not to provide unnecessary options.... @jancborchardt could maybe provide some input on this issue.

@piiskop your aggressive behaviour risks to deter contributors to develop open source software, and is not tolerated in several communities (as mtmail already clearly explained in the linked issue of Nominatim). Could you please try to have a kinder and more constructive approach?
